### PR TITLE
feat: image attachment optimisation [CRNS-474, CRNS-443]

### PR DESCRIPTION
**Pull request by vishalnarkhede**
Implementation is based on specs drawn out in following doc - https://gist.github.com/vishalnarkhede/6eed1e52b74547f71da8a5757b27c43d

## 🎯 Goal

- Optimize the size of image attachments, avatar, reply images previews rendered on UI within message list.
- Improve the designs of image attachments depending on aspect ratio of image. Please check these figma designs for details - https://www.figma.com/file/ekifwChR9tR7zRJg1QEzSM/Chat-Design-Kit---Current-version?node-id=17595%3A266053

## 🛠 Implementation details

- Plenty of utility functions have been introduced, but the main entry point or logic for generating the gallery of dynamic image sizes and grid is in [buildGallery.ts](https://github.com/GetStream/stream-chat-react-native/pull/1056/files#diff-ebeaf7b662be3f255da4db4a832c10accaf911f81f1ad5f83bf27dc308b1dab9)

## 🎨 UI Changes

<table>
    <thead>
        <tr>
            <td>Before</td>
            <td>After</td>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>
                Screenshot 2021-12-29 at 14 19 25
            </td>
            <td>
                Screenshot 2021-12-29 at 14 20 09
            </td>
        </tr>
        <tr>
            <td>
                Screenshot 2021-12-29 at 14 22 10
            </td>
            <td>
Screenshot 2021-12-29 at 14 21 49
            </td>
        </tr>
        <tr>
            <td>
                Screenshot 2021-12-29 at 14 40 29
            </td>
            <td>
                  Screenshot 2021-12-29 at 14 39 36
            </td>
        </tr>
    </tbody>
</table>
</details>

## ☑️ Checklist

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [ ] PR targets the `develop` branch
- [ ] Commits follow the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) spec
- [ ] New code is covered by tests
- [ ] Screenshots added for visual changes
- [ ] Documentation is updated

### CODE REVIEW

File: package/src/components/Message/MessageSimple/__tests__/MessageContent.test.js [hunk 1]
```diff
@@ -167,21 +167,6 @@ describe('MessageContent', () => {
     });
   });
 
-  it('renders the Gallery when image attachments exist', async () => {
```
Comment (vishalnarkhede) [inline package/src/components/Message/MessageSimple/__tests__/MessageContent.test.js:170]: This test was redundant, so removing it!!

File: .github/workflows/check-pr.yml [hunk 2]
```diff
@@ -5,7 +5,7 @@ on:
     branches:
       - develop
       - master
-      - 'v[0-9]+.[0-9]+.[0-9]+'
+      - 'v[0-9]+.[0-9]+.[0-9]+(.*)'
```
Comment (madsroskar) [inline .github/workflows/check-pr.yml:8]:
```suggestion
      - 'v[0-9]+.[0-9]+.[0-9]+(?:-(?:beta|next)(?:\.\d)?)?'
```
Think this should work

(https://regex101.com/r/Sxuygh/2)

**Comment by madsroskar** [comment]
Sorry for the extra commits Vish - wanted to get the checks running properly. You can squash my commits into one, but I don't wanna force push on your branch without having a chat first :D

File: package/src/contexts/overlayContext/OverlayProvider.tsx [hunk 3]
```diff
@@ -234,14 +234,16 @@ export const OverlayProvider = <
...
                 )}
-                <ImageGallery<At, Ch, Co, Ev, Me, Re, Us>
-                  imageGalleryCustomComponents={imageGalleryCustomComponents}
-                  imageGalleryGridHandleHeight={imageGalleryGridHandleHeight}
-                  imageGalleryGridSnapPoints={imageGalleryGridSnapPoints}
-                  numberOfImageGalleryGridColumns={numberOfImageGalleryGridColumns}
-                  overlayOpacity={overlayOpacity}
-                  visible={overlay === 'gallery'}
-                />
+                {overlay === 'gallery' && (
```
Comment (vishalnarkhede) [inline package/src/contexts/overlayContext/OverlayProvider.tsx:237]: Just to avoid rendering ImageGallery when overlay is not active

**Pull request merged**